packages/expo-cli/src/commands/client/index.js

```diff
@@ -127,7 +128,18 @@ export default program => {
           `This build request requires a valid distribution certificate.`
         );
       }
-      const pushKey = await selectPushKey(context);
+
+      let pushKey;
+      if (user) {
+        await runCredentialsManager(
+          context,
+          new SetupIosPush({ experienceName, bundleIdentifier })
+        );
+        pushKey = await context.ios.getPushKey(experienceName, bundleIdentifier);
+      } else {
+        pushKey = await new CreateIosPush().provideOrGenerate(context);
```

Comment:
push key is server-side credential so how can the anonymous user make any use of it?

Reply:
good catch, thanks!